### ISSUE TITLE
Rails/ThreeStateBooleanColumn error when checking Rails drop_table migration with boolean column

### ISSUE THREAD

**Issue by TastyPi** [rubocop/rubocop#11785]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

The cop should run without erroring

## Actual behavior

Output of `rubocop --debug`

```
For /home/graham/ophelos: configuration from /home/graham/ophelos/.rubocop.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-capybara-2.17.1/config/default.yml
Default configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-capybara-2.17.1/lib/../config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-minitest-0.29.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-minitest-0.29.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-performance-1.16.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-performance-1.16.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.19.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.19.0/config/default.yml
configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-thread_safety-0.5.1/config/default.yml
configuration from #{path}
Inheriting configuration from /home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/standard-1.26.0/config/base.yml
Inheriting configuration from /home/graham/ophelos/.rubocop_todo.yml
Inspecting 1 file
Scanning /home/graham/ophelos/db/migrate/20220728141103_drop_table_reports.rb
An error occurred while Rails/ThreeStateBooleanColumn cop was inspecting /home/graham/ophelos/db/migrate/20220728141103_drop_table_reports.rb:12:6.
undefined method `value' for nil:NilClass
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.19.0/lib/rubocop/cop/rails/three_state_boolean_column.rb:49:in `block in on_send'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.19.0/lib/rubocop/cop/rails/three_state_boolean_column.rb:85:in `three_state_boolean?'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.19.0/lib/rubocop/cop/rails/three_state_boolean_column.rb:42:in `on_send'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:141:in `each'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:137:in `each'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:158:in `on_block'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:154:in `on_class'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-ast-1.28.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cop/team.rb:98:in `investigate'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:345:in `block in inspect_file'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:344:in `each'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:344:in `flat_map'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:344:in `inspect_file'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:314:in `loop'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:164:in `block in file_offenses'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:189:in `file_offense_cache'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:163:in `file_offenses'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:154:in `process_file'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:134:in `each'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:134:in `reduce'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:134:in `each_inspected_file'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:120:in `inspect_files'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/runner.rb:73:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/command.rb:11:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli/environment.rb:18:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli.rb:118:in `run_command'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli.rb:125:in `execute_runners'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli.rb:51:in `block in run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli.rb:77:in `profile_if_needed'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/lib/rubocop/cli.rb:43:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/exe/rubocop:19:in `block in <top (required)>'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/3.2.0/benchmark.rb:311:in `realtime'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/exe/rubocop:19:in `<top (required)>'
/home/graham/.local/share/rbenv/versions/3.2.2/bin/rubocop:25:in `load'
/home/graham/.local/share/rbenv/versions/3.2.2/bin/rubocop:25:in `<top (required)>'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:58:in `load'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:58:in `kernel_load'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:23:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli.rb:492:in `exec'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli.rb:34:in `dispatch'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/cli.rb:28:in `start'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/exe/bundle:45:in `block in <top (required)>'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
/home/graham/.local/share/rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.9/exe/bundle:33:in `<top (required)>'
/home/graham/.local/share/rbenv/versions/3.2.2/bin/bundle:25:in `load'
/home/graham/.local/share/rbenv/versions/3.2.2/bin/bundle:25:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Rails/ThreeStateBooleanColumn cop was inspecting /home/graham/ophelos/db/migrate/20220728141103_drop_table_reports.rb:12:6.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.48.1 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.2.2) [x86_64-linux]
Finished in 0.7133293240000285 seconds
```

## Steps to reproduce the problem

Run RuboCop on the following file:

```ruby
# frozen_string_literal: true

# frozen_string_literal: true

class DropTableReports < ActiveRecord::Migration[7.0]
  def change
    drop_table :reports, if_exists: true do |t|
      t.boolean :download_permitted, null: false
    end
  end
end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.48.1 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-capybara 2.17.1
  - rubocop-minitest 0.29.0
  - rubocop-performance 1.16.0
  - rubocop-rails 2.19.0
  - rubocop-thread_safety 0.5.1
```
